ads/google/a4a/traffic-experiments.js

```diff
@@ -67,39 +73,45 @@ const INTERNALLY_SELECTED_ID = '2088462';
-      const selectedBranch = getPageExperimentBranch(win, experimentName);
-      addExperimentIdToElement(selectedBranch, element);
-      // Detect how page was selected into the overall experimentName.
-      if (isSetFromUrl) {
-        addExperimentIdToElement(EXTERNALLY_SELECTED_ID, element);
-      } else {
-        // Must be internally selected.
-        addExperimentIdToElement(INTERNALLY_SELECTED_ID, element);
-      }
-      // Detect whether page is on the "experiment" (i.e., use A4A rendering
-      // pathway) branch of the overall traffic experiment or it's on the
-      // "control" (i.e., use traditional, 3p iframe rendering pathway).
-      return selectedBranch == internalBranches.experiment ||
-          selectedBranch == externalBranches.experiment ||
-          selectedBranch == MANUAL_EXPERIMENT_ID;
+  if (!isGoogleAdsA4AValidEnvironment(win)) {
+    // Serving location doesn't qualify for A4A treatment
+    return false;
+  }
+
+  const isSetFromUrl = maybeSetExperimentFromUrl(win, element,
+      experimentName, externalBranches.control,
+      externalBranches.experiment, MANUAL_EXPERIMENT_ID);
+  const experimentInfo = {};
+  experimentInfo[experimentName] = internalBranches;
+  // Note: Because the same experimentName is being used everywhere here,
+  // randomlySelectUnsetPageExperiments won't add new IDs if
+  // maybeSetExperimentFromUrl has already set something for this
+  // experimentName.
+  randomlySelectUnsetPageExperiments(win, experimentInfo);
+  if (isExperimentOn(win, experimentName)) {
+    // Page is selected into the overall traffic experiment.
+    // In other words, if pre-launch serve A4A, else don't.
+    const selectedBranch = getPageExperimentBranch(win, experimentName);
+    addExperimentIdToElement(selectedBranch, element);
+    // Detect how page was selected into the overall experimentName.
+    if (isSetFromUrl) {
+      addExperimentIdToElement(EXTERNALLY_SELECTED_ID, element);
+    } else {
+      // Must be internally selected.
+      addExperimentIdToElement(INTERNALLY_SELECTED_ID, element);
     }
+    // Detect whether page is on the "experiment" (i.e., use A4A rendering
+    // pathway) branch of the overall traffic experiment or it's on the
+    // "control" (i.e., use traditional, 3p iframe rendering pathway).
+    const selected = selectedBranch == internalBranches.experiment ||
+                     selectedBranch == externalBranches.experiment ||
+                     selectedBranch == MANUAL_EXPERIMENT_ID;
+    return (selected == !hasLaunchedDoubleClick(element)); // TODO Not necessarily DoubleClick!
+  } else {
+    // Page is not selected into the overall traffic experiment.
+    // In other words, if post-launch serve A4A, else don't
+    return hasLaunchedDoubleClick(element); // TODO Not necessarily DoubleClick!
   }
-  // Serving location doesn't qualify for A4A treatment or page is not in the
-  // traffic experiment.
-  return false;
```

Comment:
One hopes that the Closure compiler would catch unreachable code.  But I've been bitten by things that Closure doesn't do.  Try it and see if the build cycle complains.  (Try `gulp dist`.)